src/codehelp/nodesearch.vala

```diff
@@ -595,7 +595,7 @@ class Vls.NodeSearch : Vala.CodeVisitor {
 
     public override void visit_switch_statement (Vala.SwitchStatement stmt) {
         if (!seen.add (stmt)) return;
-        if (this.match (stmt))
+        if (this.match (stmt) || new Range.from_sourceref (stmt.source_reference).contains (pos))
```

Comment:
See if you can update `match()` instead. I'd prefer to have all the logic in one place.

Reply:
Just do something like this in match:

```vala
... match(...) {
  ...
  if (node is Vala.SwitchStatement) {
    return new Range.from_sourceref (( Vala.SwitchStatement)node).source_reference).contains (pos)
  }
}
```